src/ripple/app/consensus/RCLValidations.cpp

```diff
@@ -154,68 +154,66 @@ handleNewValidation(
     std::shared_ptr<STValidation> const& val,
     std::string const& source)
 {
-    PublicKey const& signingKey = val->getSignerPublic();
-    uint256 const& hash = val->getLedgerHash();
+    auto const& signingKey = val->getSignerPublic();
+    auto const& hash = val->getLedgerHash();
+    auto const seq = val->getFieldU32(sfLedgerSequence);
 
     // Ensure validation is marked as trusted if signer currently trusted
     auto masterKey = app.validators().getTrustedKey(signingKey);
+
     if (!val->isTrusted() && masterKey)
         val->setTrusted();
 
     // If not currently trusted, see if signer is currently listed
     if (!masterKey)
         masterKey = app.validators().getListedKey(signingKey);
 
-    RCLValidations& validations = app.getValidations();
-    beast::Journal const j = validations.adaptor().journal();
-
-    auto dmp = [&](beast::Journal::Stream s, std::string const& msg) {
-        std::string id = toBase58(TokenType::NodePublic, signingKey);
-
-        if (masterKey)
-            id += ":" + toBase58(TokenType::NodePublic, *masterKey);
-
-        s << (val->isTrusted() ? "trusted" : "untrusted") << " "
-          << (val->isFull() ? "full" : "partial") << " validation: " << hash
-          << " from " << id << " via " << source << ": " << msg << "\n"
-          << " [" << val->getSerializer().slice() << "]";
-    };
+    auto& validations = app.getValidations();
 
     // masterKey is seated only if validator is trusted or listed
-    if (masterKey)
+    auto const outcome =
```

Comment:
It's been a while since I've looked at this, can you clarify this comment being unchanged?

> masterKey is seated only if validator is trusted or listed

Your comment https://github.com/ripple/rippled/pull/3778#issue-578184333 suggested we only previously checked trusted validations.

The

Reply:
This is somewhat confusing, and I understand why you'd call this out. The comment is still accurate (sorta-kinda): we have a validation signed with some key K which _may_ be an ephemeral key or it may be a "direct" key (i.e. set via `[validation_seed]`). If it's signed by a manifest **AND** we know the manifest, we want to retrieve the master key and use that, otherwise, we use the ephemeral key.

Operating with "direct" keys (i.e. without a manifest) should be deprecated and, eventually, not allowed. And, if we receive a validation for an ephemeral key whose manifest the server does not have, we ought to just ignore it, especially post 1.7 that fixed manifest propagation.